processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -385,19 +385,20 @@ private void initLongData()
           if (hasLowerBound()) {
             final Long lowerLong = GuavaUtils.tryParseLong(lower);
             if (lowerLong == null) {
-              matchesAnything = false;
-              return;
+              // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
+              hasLowerLongBoundVolatile = false;
+            } else {
+              hasLowerLongBoundVolatile = true;
+              lowerLongBoundVolatile = lowerLong;
             }
-
-            hasLowerLongBoundVolatile = true;
-            lowerLongBoundVolatile = lowerLong;
           } else {
             hasLowerLongBoundVolatile = false;
```

Comment:
@leventov what you said is true, but I don't see harm in marking them all volatile and keeping things simple here to understand. is there a catch?